staging/src/k8s.io/kube-aggregator/pkg/controllers/openapiv3/aggregator/aggregator.go

```diff
@@ -73,29 +77,60 @@ func (s *specProxier) GetAPIServiceNames() []string {
 }
 
 // BuildAndRegisterAggregator registered OpenAPI aggregator handler. This function is not thread safe as it only being called on startup.
-func BuildAndRegisterAggregator(downloader Downloader, delegationTarget server.DelegationTarget, pathHandler common.PathHandlerByGroupVersion) (SpecProxier, error) {
+func BuildAndRegisterAggregator(downloader Downloader, delegationTarget server.DelegationTarget, webServices []*restful.WebService, config *common.Config, pathHandler common.PathHandlerByGroupVersion) (SpecProxier, error) {
 	s := &specProxier{
 		apiServiceInfo: map[string]*openAPIV3APIServiceInfo{},
 		downloader:     downloader,
 	}
 
 	i := 1
+
+	if len(webServices) > 0 && config != nil {
+		// first register aggregator webServices (eg. apiregistration.k8s.io)
+		aggregatorWebServicesHandler := handler3.NewOpenAPIService()
+
+		grouped := make(map[string][]*restful.WebService)
+
+		for _, t := range webServices {
+			// Strip the "/" prefix from the name
+			gvName := t.RootPath()[1:]
+			grouped[gvName] = []*restful.WebService{t}
+		}
+
+		for gv, ws := range grouped {
+			spec, err := builder3.BuildOpenAPISpecFromRoutes(restfuladapter.AdaptWebServices(ws), config)
+			if err != nil {
+				klog.Errorf("Failed to build OpenAPI v3 for group %s, %q", gv, err)
+
+			}
+			aggregatorWebServicesHandler.UpdateGroupVersion(gv, spec)
+		}
+		apiServiceName := fmt.Sprintf(localDelegateChainNamePattern, i)
```

Comment:
This isn't really a delegation handler in the traditional sense. The aggregator serves the path, can we use a constant name for this to not affect the local delegation numbers?

Reply:
renamed to `k8s_internal_local_kube_aggregator_types` to at least have the same prefix as the delegation handlers